tick/simulation/SCCS.py

```diff
@@ -0,0 +1,379 @@
+        """Simulates features, either `infinite` or `short` exposures."""
+        if self.exposure_type == "infinite":
+            sim_feat = lambda: self._sim_infinite_exposures(self.sparse)
+        elif self.exposure_type == "short":
+            sim_feat = lambda: self._sim_short_exposures(self.sparse)
+
+        return [sim_feat() for i in range(n_samples)]
+
+    def _sim_short_exposures(self, sparse):
+        features = np.random.randint(2,
+                                     size=(self.n_intervals, self.n_features),
+                                     ).astype("float64")
+        if sparse:
+            features = csr_matrix(features, dtype="float64")
+        return features
+
+    def _sim_infinite_exposures(self, sparse):
+        if not sparse:
+            raise ValueError("'infinite' exposures can only be simulated as \
+            sparse feature matrices")
+        # Select features for which there is exposure
+        if self.n_features > 1:
+            n_exposures = np.random.randint(1, self.n_features, 1)
+            cols = np.random.choice(np.arange(self.n_features, dtype="int64"),
+                                    n_exposures, replace=False)
+        else:
+            n_exposures = 1
+            cols = np.array([0])
+        # choose exposure start
+        rows = np.random.randint(self.n_intervals, size=n_exposures)
+        # build sparse matrix
+        data = np.ones_like(cols, dtype="float64")
+        exposures = csr_matrix((data, (rows, cols)),
+                               shape=(self.n_intervals, self.n_features),
+                               dtype="float64")
+        return exposures
+
+    def _simulate_outcomes(self, features, censoring):
+        features = LongitudinalFeaturesLagger(n_lags=self.n_lags).\
+            fit_transform(features, censoring)
+
+        if self.distribution == "poisson":
+            y = self._simulate_outcome_from_poisson(features, self.coeffs,
+                                                    self.first_tick_only)
+        else:
+            y = self._simulate_outcome_from_multi(features, self.coeffs,)
+        return y
+
+    @staticmethod
+    def _simulate_outcome_from_multi(features, coeffs):
+        inner_product = [f.dot(coeffs) for f in features]
+
+        def simulate(inner_prod):
+            inner_prod -= inner_prod.max()
+            probabilities = np.exp(inner_prod) / \
+                            np.sum(np.exp(inner_prod))
+            y = np.random.multinomial(1, probabilities)
+            return y.astype("uint64")
+
+        return [simulate(i) for i in inner_product]
+
+    @staticmethod
+    def _simulate_outcome_from_poisson(features, coeffs, first_tick_only=True):
+        inner_product = [f.dot(coeffs) for f in features]
+
+        def simulate(inner_prod):
+            intercept = -inner_prod.max()
+            intensities = np.exp(intercept + inner_prod)
+            ticks = np.random.poisson(lam=intensities)
```

Comment:
intensities.max() should not be too large... otherwise raise a ValueError saying that intensity is too large and Poisson simulation will fail because of overflows